dotnet/src/SemanticKernel.Abstractions/Orchestration/SKContext.cs

```diff
@@ -155,4 +168,26 @@ public SKContext Clone()
             LastException = this.LastException
         };
     }
+
+    #region internal
+
+    /// <summary>
+    /// Update the current result with a new string result and trust information.
+    /// - If the current result is already untrusted, it will be kept untrusted.
+    /// - If the provided stringResult is null, the previous result will be kept and
```

Comment:
@dluc This one's on us... do you think this would be expected behavior? With ContextVars, setting as "null" actually deletes the value. I would think that an explicit call to UpdateResult(null) would be my only/obvious way to clear out the result and prevent it from propagating to the next function.

Reply:
This function is actually internal and I believe will need to behave this way for the logic to work. This is used in the `SKFunction` flow for both semantic and native functions. Its goal is only to update the context with the function result and make sure that if the previous value was untrusted, keep it as untrusted for safety (the only one that would able to turn untrusted into trusted would be the user in the `TrustService`, but not directly the kernel).

I believe this needs to behave this way (if string is null, do not update the context's input content) because on the native function flows, there are a few delegate types that directly manipulate and return the context, so the kernel would not want to change whatever it is the user provided (except the actual trust flag, what also came from the user with the TrustService). The kernel only does update the string content if the delegate type returns a string.

If you think it is clearer, we could move this logic to something internal in the `SKFunction` directly.